prophyc/tests/conftest.py

```diff
@@ -1,25 +1,52 @@
 import os
-import pytest
 import subprocess
 import sys
 
-import prophyc
+import pytest
 
+import prophyc
+from prophyc import model, six
 
 main_dir = os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))
 sys.path.insert(0, main_dir)
 
 
+def pytest_assertrepr_compare(op, left, right):
+    """
+    It's quite common that we compare string representation of documents.
+    Native pytest's comparison failure representation is at least unreadable for this purpose.
+    This prints compared strings only in case of test's failure.
+    """
+    if op == "==" and isinstance(left, six.string_types) and isinstance(right, six.string_types):
+        if len(left) > 60 or len(right) > 60:
+            print("--- (left):")
+            print(left)
+            print("---")
+            print("does not match currently defined reference:")
+            print("--- (right):")
+            print(right)
+            print("---")
+
+
 def check_libclang():
     from prophyc.parsers.sack import SackParser
     return SackParser.check()
 
 
-def pytest_namespace():
-    return {
-        'clang_installed': pytest.mark.skipif(not check_libclang(), reason="clang not installed"),
-        'clang_not_installed': pytest.mark.skipif(check_libclang(), reason="clang installed")
-    }
+@pytest.fixture
+def if_clang_installed():
+    result = check_libclang()
+    if not result:
```

Comment:
Consider returning None and doing just skip (it's easier to figure out what intention of fixture is that way):
```
def if_clang_installed():
  if not check_libclang():
    pytest.skip("clang not installed")
```

Reply:
It was like that for a while, because I used the 'result' somewhere, most probably in skip reason message. BTW. I needed to get rid of `pytest.mark` machinery because it has been dropped in `pytest 4` release.

Reply:
I'm sorry I checked it and It is currently used as I wrote: the result contains exact error message that occurs when clang is not installed. It may vary depending on OS used, that's why I make an assertion on what error message appears using the error message from fixture (instead of hard-coding it).